Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -240,8 +240,8 @@ void GroupDetectors2::execEvent() {
   }
 
   // Set all X bins on the output
-  auto XValues = HistogramData::BinEdges(inputWS->readX(0));
-  outputWS->setAllX(XValues);
+  auto binEdges = inputWS->binEdges(0);
+  outputWS->setAllX(binEdges);
```

Comment:
Even shorter: `outputWS->setAllX(inputWS->binEdges(0));`, no need for the temporary.